x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -744,74 +744,57 @@ public void invalidateRefreshToken(String refreshToken, ActionListener<TokensInv
     }
 
     /**
-     * Invalidates all access tokens and all refresh tokens of a given {@code realmName} and/or of a given
-     * {@code username} so that they may no longer be used
+     * Invalidates all access and refresh tokens for a given {@code realmName} and/or of a given
+     * {@code username} so that they may no longer be usable.
      *
      * @param realmName the realm of which the tokens should be invalidated
      * @param username the username for which the tokens should be invalidated
+     * @param filter    An optional {@code Predicate} to further test and filter the tokens to invalidate.
+     *                  The predicate tests the token doc source.
      * @param listener  the listener to notify upon completion
      */
-    public void invalidateActiveTokensForRealmAndUser(
+    public void invalidateActiveTokens(
         @Nullable String realmName,
         @Nullable String username,
+        @Nullable Predicate<Map<String, Object>> filter,
```

Comment:
Added optional new parameter to make it reusable in the `TransportSamlInvalidateSessionAction` context.